### ISSUE TITLE
Add support for enforcing type of `_value_` in Enum class if it is declared

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7030]
If an Enum class explicitly declares the type of the `_value_` attribute, pyright should honor it and enforce type consistency.

```python
from enum import Enum


class Color(Enum):
    _value_: int
    RED = 1 # OK
    GREEN = "green"  # Should be type error

class Planet(Enum):
    _value_: str

    def __init__(self, value: int, mass: float, radius: float):
        self._value_ = value # Shoudl be type error

    MERCURY = (1, 3.303e+23, 2.4397e6)
```

**Comment by srivamsidandu** [comment]
Hi i want to work on this issue 
but cannot find in which file this code  lies in

i want to introduce a new base class  that includes the _value_: type attribute. The  function is then used to iterate over the members of the Enum class and check if the types match. If there's a type mismatch, it raises a TypeError.

assign me this issue , so that i can rectify it

**Comment by erictraut** [comment]
> Hi i want to work on this issue

Please don't. Pyright is a complex code base — especially its core type evaluator. It's much easier and faster for me to make changes like this.

**Comment by erictraut** [comment]
This will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.348, which was just published. It will also be included in a future release of pylance.
